src/screens/UserManagement/UserRevamp/UserManagementHelper.res

```diff
@@ -228,6 +228,130 @@ module SwitchMerchantForUserAction = {
   }
 }
 
+module OMPViewBaseComp = {
+  @react.component
+  let make = (~userModuleEntity: UserManagementTypes.userModuleTypes, ~arrow) => {
+    let (_, getNameForId) = OMPSwitchHooks.useOMPData()
+
+    let arrowUpClass = "rotate-0 transition duration-[250ms] opacity-70"
+    let arrowDownClass = "rotate-180 transition duration-[250ms] opacity-70"
+
+    let displayName = switch userModuleEntity {
+    | #Default => "My Team"
+    | _ => userModuleEntity->getNameForId
+    }
+
+    let truncatedDisplayName = if displayName->String.length > 15 {
+      <HSwitchOrderUtils.EllipsisText
```

Comment:
If generic component please move it to HSwitchUtils.res

Reply:
It was causing cyclic dependency, might have to shift few other components to achieve the same

Reply:
shifted components to remove the cyclic dependencies